kratos/includes/properties.h

```diff
@@ -206,51 +212,71 @@ class Properties : public IndexedObject
     }
 
     template<class TVariableType>
-    typename TVariableType::Type const& GetValue(const TVariableType& rV) const
+    typename TVariableType::Type const& GetValue(const TVariableType& rVariable) const
     {
-        return mData.GetValue(rV);
+
+        return mData.GetValue(rVariable);
     }
 
     template<class TVariableType>
-    typename TVariableType::Type& GetValue(const TVariableType& rV, NodeType& rThisNode)
+    typename TVariableType::Type& GetValue(const TVariableType& rVariable, NodeType& rThisNode)
     {
-        if(mData.Has(rV))
-            return mData.GetValue(rV);
-
-        return rThisNode.GetValue(rV);
+        if (mData.Has(rVariable))
+            return mData.GetValue(rVariable);
+        return rThisNode.GetValue(rVariable);
     }
 
     template<class TVariableType>
-    typename TVariableType::Type const& GetValue(const TVariableType& rV, NodeType const& rThisNode) const
+    typename TVariableType::Type const& GetValue(const TVariableType& rVariable, NodeType const& rThisNode) const
     {
-        if(mData.Has(rV))
-            return mData.GetValue(rV);
-
-        return rThisNode.GetValue(rV);
+        if (mData.Has(rVariable))
+            return mData.GetValue(rVariable);
+        return rThisNode.GetValue(rVariable);
     }
 
     template<class TVariableType>
-    typename TVariableType::Type& GetValue(const TVariableType& rV, NodeType& rThisNode, IndexType SolutionStepIndex)
+    typename TVariableType::Type& GetValue(const TVariableType& rVariable, NodeType& rThisNode, IndexType SolutionStepIndex)
     {
-        if(mData.Has(rV))
-            return mData.GetValue(rV);
+        if (mData.Has(rVariable))
+            return mData.GetValue(rVariable);
+        return rThisNode.GetValue(rVariable, SolutionStepIndex);
+    }
 
-        return rThisNode.GetValue(rV, SolutionStepIndex);
+    template<class TVariableType>
+    typename TVariableType::Type const& GetValue(const TVariableType& rVariable, NodeType const& rThisNode, IndexType SolutionStepIndex) const
+    {
+        if (mData.Has(rVariable))
+            return mData.GetValue(rVariable);
+        return rThisNode.GetValue(rVariable, SolutionStepIndex);
     }
 
+    /*
+    Custom GetValue in which we check the Accessor
+    */
     template<class TVariableType>
-    typename TVariableType::Type const& GetValue(const TVariableType& rV, NodeType const& rThisNode, IndexType SolutionStepIndex) const
+    typename TVariableType::Type GetValue(const TVariableType& rVariable, const GeometryType &rGeometry, const Vector& rShapeFunctionVector, const ProcessInfo &rProcessInfo) const
     {
-        if(mData.Has(rV))
-            return mData.GetValue(rV);
+        auto it_value = mAccessors.find(rVariable.Key());
+        if (it_value != mAccessors.end()) {
+            return (it_value->second)->GetProperty(rVariable, *this, rGeometry, rShapeFunctionVector, rProcessInfo);
+        } else {
+            return mData.GetValue(rVariable);
```

Comment:
Maybe we should add a Warning if not accesor found

Reply:
well... This is a decision we have to make

Reply:
https://www.boost.org/doc/libs/1_81_0/libs/unordered/doc/html/unordered.html#benchmarks_boostunordered_flat_map I still think we should also discuss about using this instead